spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -81,6 +83,10 @@
 
 	private volatile AbstractMessageChannelMetrics channelMetrics = new DefaultMessageChannelMetrics();
 
+	public AbstractMessageChannel() {
+		this.interceptors = new ChannelInterceptorList(logger);
+	}
```

Comment:
Yes; that's how I had it at first, but I am not sure if it is safe to use the super field in the initialization code whereas in the ctor, we know that super() has completed. If you know for sure it's safe we can revert.